src/components/ArticleList.jsx

```diff
@@ -1,37 +1,58 @@
 import ArticleCard from "./ArticleCard"
-import axios from "axios"
 import React from "react"
 import {useEffect, useState} from "react"
 import api from "./Api"
+import { useNavigate, useParams } from "react-router-dom"
 
 export default function ArticleList () {
-    const [articlesList, setArticlesList] = useState([])
-    const [loading, setLoading] = useState(true)
+  const { topic } = useParams()
+  const [articlesList, setArticlesList] = useState([])
+  const [loading, setLoading] = useState(true)
+  const navigate = useNavigate()
 
 
-    useEffect(() => {
-        api.get("/articles")
-          .then((response) => {
-            setArticlesList(response.data.articles);
-            setLoading(false);
+  useEffect(() => {
+    const fetchArticles = () => {
```

Comment:
You don't need this function wrapped around the functionality, you can remove it and also remove line 30 and the call to get articles will just trigger when the useeffect triggers